hub/src/server/drivers/AzDriver.ts

```diff
@@ -187,7 +189,7 @@ class AzDriver implements DriverModel, DriverModelTestMethods {
     const maxBuffers = 1
 
     try {
-      await azure.uploadStreamToBlockBlob(
+      const uploadResult = await azure.uploadStreamToBlockBlob(
         azure.Aborter.none, args.stream,
         blockBlobURL, bufferSize, maxBuffers, {
```

Comment:
I believe Azure blob uploads support If-Match etag conditions directly, see: https://docs.microsoft.com/en-us/javascript/api/%40azure/storage-blob/blockblobuploadstreamoptions?view=azure-node-latest#conditions

We should implement this at least for Azure, since that'll have the most impact on the default user Gaia hub.

Reply:
Yes, I believe that's correct too. We included an optional `if-match` condition in the `DriverModel`s `performWrite` option type for this purpose. Was planning on implementing this in a followup PR, but perhaps it should be included in this one.